src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
```

Comment:
1. It is better to avoid names with one letter, in this case you can use the destructuring
https://mate-academy.github.io/style-guides/javascript.html#51-use-object-destructuring-when-accessing-and-using-multiple-properties-of-an-object
2. The assignment says not to use forEach, but instead you can use reduce to find the sum 
https://mate-academy.github.io/style-guides/javascript.html#11-iterators-and-generators